### PR TITLE
041 add authorisation

### PR DESCRIPTION
**Pull request by rackie-pascua**

### CODE REVIEW

**Review by JamieMcConnellKainos — DISMISSED** [review]
looks good

**Review by JamieMcConnellKainos — COMMENTED** [review]
Very minor point but maybe remove commented unused code from AuthMiddlewareTest.ts ?

**Review by JamieMcConnellKainos — APPROVED** [review]
Approved

File: src/middleware/AuthMiddleware.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { jwtDecode } from "jwt-decode";
+import { JwtToken, UserRole } from "../models/JwtToken";
+import express from "express";
+
+export const allowRoles = (allowedRoles: UserRole[]) => {
+    return (req: express.Request, res: express.Response, next: express.NextFunction) => {
+        if (!req.session.token) {
+            return res.status(401).send('Not logged in');
```
Comment (boydstepkainos) [inline src/middleware/AuthMiddleware.ts:8]: Requirement is non-logged in users are redirected to login page

File: src/middleware/AuthMiddleware.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        }
+
+        const decodedToken: JwtToken = jwtDecode(req.session.token);
+        if (!allowedRoles.includes(decodedToken.Role)) {
+            return res.status(403).send('User role not authorised for this action');
```
Comment (boydstepkainos) [inline src/middleware/AuthMiddleware.ts:13]: Do we need to redirect anywhere if they are not authorised?

File: src/middleware/AuthMiddleware.ts (hunk 1 above)
Comment (rackie-pascua) [reply src/middleware/AuthMiddleware.ts]: done

**Review by ashnabraham — DISMISSED** [review]
:)

**Review by mark-mcilroykainos — APPROVED** [review]

**Review by shaunganley — APPROVED** [review]

**Pull request merged**